docs/keycodes.md

```diff
@@ -0,0 +1,299 @@
+|`KC.EXCLAIM`            |`KC.EXLM`          |`!`                |
+|`KC.AT`                 |                   |`@`                |
+|`KC.HASH`               |                   |`#`                |
+|`KC.DOLLAR`             |`KC.DLR`           |`$`                |
+|`KC.PERCENT`            |`KC.PERC`          |`%`                |
+|`KC.CIRCUMFLEX`         |`KC.CIRC`          |`^`                |
+|`KC.AMPERSAND`          |`KC.AMPR`          |`&`                |
+|`KC.ASTERISK`           |`KC.ASTR`          |`*`                |
+|`KC.LEFT_PAREN`         |`KC.LPRN`          |`(`                |
+|`KC.RIGHT_PAREN`        |`KC.RPRN`          |`)`                |
+|`KC.UNDERSCORE`         |`KC.UNDS`          |`_`                |
+|`KC.PLUS`               |                   |`+`                |
+|`KC.LEFT_CURLY_BRACE`   |`KC.LCBR`          |`{`                |
+|`KC.RIGHT_CURLY_BRACE`  |`KC.RCBR`          |`}`                |
+|`KC.PIPE`               |                   |<code>&#124;</code>|
+|`KC.COLON`              |`KC.COLN`          |`:`                |
+|`KC.DOUBLE_QUOTE`       |`KC.DQUO`, `KC.DQT`|`"`                |
+|`KC.LEFT_ANGLE_BRACKET` |`KC.LABK`, `KC.LT` |`<`                |
+|`KC.RIGHT_ANGLE_BRACKET`|`KC.RABK`, `KC.GT` |`>`                |
+|`KC.QUESTION`           |`KC.QUES`          |`?`                |
+
+
+## [Internal Keycodes]
+
+|Key                    |Description                                                          |
+|-----------------------|---------------------------------------------------------------------|
+|`KC.RESET`             |Put the keyboard into DFU mode for flashing                          |
+|`KC.DEBUG`             |Toggle debug mode                                                    |
+|`KC.GESC`              |Escape when tapped, <code>&#96;</code> when pressed with Shift or GUI|
+|`KC.LEAD`              |The [Leader key]                                                     |
+|`KC.UC_MODE_NOOP`      |Sets UnicodeMode to NOOP                                             |
+|`KC.UC_MODE_LINUX`     |Sets UnicodeMode to NOOP                                             |
+|`KC.UC_MODE_MACOS`     |Sets UnicodeMode to NOOP                                             |
+|`KC.UC_MODE_WINC`      |Sets UnicodeMode to NOOP                                             |
+|`KC.MACRO_SLEEP_MS(ms)`|Sleeps in a macro. Check MACROS for more information.                |
+
+
+## [Layer Switching]
+
+|Key         |Description                                                                  |
+|-----------------|------------------------------------------------------------------------|
+|`KC.DF(layer)`      |Switches the default layer                                           |
+|`KC.MO(layer)`      |Momentarily activates layer, switches off when you let go            |
+|`KC.LM(layer, mod)` |As `MO(layer)` but with `mod` active                                 |
```

Comment:
Do we actually have this support? I have no idea.

Reply:
We support all layers. I handled those.